lib/screens/v2/widget_instance/elevator_status.ex

```diff
@@ -0,0 +1,403 @@
+  # @max_height_station_heading 0
+  # @max_height_elevator_description 0
+  # @max_height_row_separator 0
+
+  def parent_station_id(%__MODULE__{
+        screen: %Screen{
+          app_params: %PreFare{
+            elevator_status: %ElevatorStatus{parent_station_id: parent_station_id}
+          }
+        }
+      }) do
+    parent_station_id
+  end
+
+  defp get_active_at_home_station(%__MODULE__{alerts: alerts} = t) do
+    alerts
+    |> Enum.filter(&active_at_home_station?(&1, t))
+  end
+
+  defp get_active_elsewhere(
+         %__MODULE__{
+           alerts: alerts
+         } = t
+       ) do
+    alerts
+    |> Enum.filter(&active_elsewhere?(&1, t))
+    |> Enum.sort_by(
+      fn %Alert{informed_entities: entities} -> entities end,
+      &sort_elsewhere(&1, &2, t)
+    )
+  end
+
+  defp get_upcoming_at_home_station(%__MODULE__{alerts: alerts} = t) do
+    alerts
+    |> Enum.filter(&upcoming_at_home_station?(&1, t))
+  end
+
+  defp get_active_on_connecting_lines(%__MODULE__{alerts: alerts} = t) do
+    alerts
+    |> Enum.filter(&active_on_connecting_lines?(&1, t))
+    |> Enum.sort_by(
+      fn %Alert{informed_entities: entities} -> entities end,
+      &sort_elsewhere(&1, &2, t)
+    )
+  end
+
+  defp active_at_home_station?(
+         %Alert{effect: :elevator_closure, informed_entities: entities} = alert,
+         %__MODULE__{
+           now: now
+         } = t
+       ) do
+    Alert.happening_now?(alert, now) &&
+      Enum.any?(entities, fn entity ->
+        entity.stop == parent_station_id(t)
+      end)
+  end
+
+  defp active_elsewhere?(
+         %Alert{effect: :elevator_closure, informed_entities: entities} = alert,
+         %__MODULE__{
+           now: now
+         } = t
+       ) do
+    stations = get_stations_from_entities(entities)
+
+    Alert.happening_now?(alert, now) &&
+      Enum.any?(stations, fn station ->
+        station != parent_station_id(t)
+      end)
+  end
+
+  defp upcoming_at_home_station?(
+         %Alert{effect: :elevator_closure, informed_entities: entities} = alert,
+         %__MODULE__{now: now} = t
+       ) do
+    not Alert.happening_now?(alert, now) &&
+      Enum.any?(entities, fn entity ->
+        entity.stop == parent_station_id(t)
+      end)
+  end
+
+  defp active_on_connecting_lines?(
+         %Alert{effect: :elevator_closure, informed_entities: entities} = alert,
+         %__MODULE__{now: now, stop_sequences: stop_sequences} = t
+       ) do
+    stations = get_stations_from_entities(entities)
+
+    Alert.happening_now?(alert, now) &&
+      Enum.any?(stations, fn station ->
+        Enum.any?(stop_sequences, fn stop_sequence ->
+          station in stop_sequence and station != parent_station_id(t)
+        end)
+      end)
+  end
+
+  defp sort_elsewhere(e1, _e2, %__MODULE__{stop_sequences: stop_sequences}) do
+    stations = get_stations_from_entities(e1)
+
+    Enum.any?(stations, fn station ->
+      station in stop_sequences
```

Comment:
Good catch. I think I fixed it in a different function but not here. Regardless, I flattened the list so that this code will actually work. I thought it was more readable than two nest `any?` calls.